src/components/app/data/utils.js

```diff
@@ -747,3 +747,54 @@ export function isEnrollmentUpgradeable(enrollment) {
   const canUpgradeToVerifiedEnrollment = enrollment.mode === COURSE_MODES_MAP.AUDIT && !isEnrollByLapsed;
   return canUpgradeToVerifiedEnrollment;
 }
+
+export function determineAllocatedCourseRuns({
+  redeemableLearnerCreditPolicies,
+  courseKey,
+}) {
+  const { learnerContentAssignments } = redeemableLearnerCreditPolicies;
+  if (learnerContentAssignments.hasAllocatedAssignments) {
+    let allocatedCourseRunAssignments = learnerContentAssignments.allocatedAssignments.filter(
+      (assignment) => assignment?.isAssignedCourseRun,
+    );
+    if (courseKey) {
+      allocatedCourseRunAssignments = allocatedCourseRunAssignments.filter(
+        (assignment) => assignment?.parentContentKey === courseKey,
+      );
+    }
+    const allocatedCourseRunAssignmentKeys = allocatedCourseRunAssignments.map(assignment => assignment.contentKey);
+    const hasAssignedCourseRuns = allocatedCourseRunAssignmentKeys.length > 0;
+    const hasMultipleAssignedCourseRuns = allocatedCourseRunAssignmentKeys.length > 1;
+    return {
+      allocatedCourseRunAssignmentKeys,
+      allocatedCourseRunAssignments,
+      hasAssignedCourseRuns,
+      hasMultipleAssignedCourseRuns,
+    };
+  }
+  return {
+    allocatedCourseRunAssignmentKeys: [],
+    allocatedCourseRunAssignments: [],
+    hasAssignedCourseRuns: false,
+    hasMultipleAssignedCourseRuns: false,
+  };
+}
```

Comment:
Spent some time making this a hook that sourced its own `courseKey` and `redeemableLearnerCreditPolicies` ran into an issue attempting to use this in the `couresLoader`. B/c the `courseLoader` is not a functional react component, I could not utilize this function as a hook. The tradeoff of using a hook vs a function was not including the performance enhancements of the `can-redeem` call, so I opted to retain the performance enhancements of `can-redeem` and keep it as a function.

One improvement is the return of the `allocatedCourseRunAssignments` value which returns the metadata of the  course run based allocations versus just the course keys.